contracts/JBTieredNFTRewardDelegate.sol

```diff
@@ -0,0 +1,396 @@
+  // ------------------------- external views -------------------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice 
+    The reward tiers.
+
+    @return The tiers. 
+  */
+  function tiers() external view override returns (JBNFTRewardTier[] memory) {
+    return _tiers;
+  }
+
+  /** 
+    @notice 
+    The total supply of issued NFTs from all tiers.
+
+    @return supply The total number of NFTs between all tiers.
+  */
+  function totalSupply() external view override returns (uint256 supply) {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = _tiers.length;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier memory _tier;
+
+    for (uint256 _i; _i < _numTiers; ) {
+      // Set the tier being iterated on.
+      _tier = _tiers[_i];
+
+      // Increment the total supply with the amount used already.
+      supply += _tier.initialAllowance - _tier.remainingAllowance;
+
+      unchecked {
+        ++_i;
+      }
+    }
+  }
+
+  /** 
+    @notice 
+    The total number of tokens with the given ID.
+
+    @return Either 1 if the token has been minted, or 0 if it hasnt.
+  */
+  function tokenSupply(uint256 _tokenId) external view override returns (uint256) {
+    return _ownerOf[_tokenId] != address(0) ? 1 : 0;
+  }
+
+  /** 
+    @notice 
+    The total number of tokens owned by the given owner. 
+
+    @param _owner The address to check the balance of.
+
+    @return The number of tokens owners by the owner.
+  */
+  function totalOwnerBalance(address _owner) external view override returns (uint256) {
+    return _balanceOf[_owner];
+  }
+
+  /** 
+    @notice 
+    The total number of tokens with the given ID owned by the given owner. 
+
+    @param _owner The address to check the balance of.
+    @param _tokenId The ID of the token to check the owner's balance of.
+
+    @return Either 1 if the owner has the token, or 0 if it does not.
+  */
+  function ownerTokenBalance(address _owner, uint256 _tokenId)
+    external
+    view
+    override
+    returns (uint256)
+  {
+    return _ownerOf[_tokenId] == _owner ? 1 : 0;
+  }
+
+  //*********************************************************************//
+  // -------------------------- public views --------------------------- //
+  //*********************************************************************//
+
+  /** 
+    @notice
+    The tier number of the provided token ID. 
+
+    @dev
+    Tier's are 1 indexed from the `tiers` array, meaning the 0th element of the array is tier 1.
+
+    @param _tokenId The ID of the token to get the tier number of. 
+
+    @return The tier number of the specified token ID.
+  */
+  function tierNumberOfToken(uint256 _tokenId) public view override returns (uint256) {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = _tiers.length;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier memory _tier;
```

Comment:
wondering if storage here is not cheaper then (ref to the key/pointer to _tiers[i], then the first sload *should* push the thing in stack)

this is a todo for after unit tests are done

Reply:
noted!